feat(docs-infra): update CLI templates to match changes with new parser and folder structure

alan-agius4 opened the pull request:
The underlying parser that the CLI uses changed which caused minor changes in the generated JSON helps especially for subcommands.

The folder structure of the Angular CLI repo also changed slightly.

More context: https://github.com/angular/angular-cli/pull/22778

alan-agius4 committed:
feat(docs-infra): update CLI templates to match changes with new parser and folder structure

The underlying parser that the CLI uses changed which caused minor changes in the generated JSON helps especially for subcommands.

The folder structure of the Angular CLI repo also changed slightly.

More context: https://github.com/angular/angular-cli/pull/22778

alan-agius4 commented:
CI will fail until we update the CLI SHA. Can I get a preemptive review?

alan-agius4 committed:
feat(docs-infra): remove hidden commands processor

Hidden commands are no longer included in help JSON output.

alan-agius4 committed:
fixup! feat(docs-infra): update CLI templates to match changes with new parser and folder structure

gkalpak commented on aio/tools/transforms/cli-docs-package/processors/processCliCommands.js:
```diff
@@ -29,6 +43,31 @@ module.exports = function processCliCommands(createDocMessage) {
+  // Handle required and optional command line args.
+  // Wrap them in a <var> element.
+  const commandUsageHtmlString = usage
+    .replace(/>/g, '&gt;&gt;/var>')
```
Shouldn't this be:

```suggestion
    .replace(/>/g, '&gt;/var>&gt;')
```

alan-agius4 replied:
You are right

gkalpak commented on aio/tools/transforms/templates/cli/lib/cli.html:
```diff
@@ -21,17 +20,9 @@
   {% for option in arguments %}
   <tr class="cli-option">
-    <td><code class="cli-option-syntax no-auto-link">&lt;<var>{$ option.name $}</var>&gt;</code></td>
+    <td><code class="cli-option-syntax no-auto-link"><var>{$ option.name $}</var></code></td>
```
OOC, why this change?

alan-agius4 replied:
Because `<…>`, means it’s a required positional argument which in this case it can be wrong and should be displayed as part of the description as `<>` or  `[]` will be displayed in the usage.

gkalpak commented on aio/tools/transforms/templates/cli/cli-command.template.html:
```diff
@@ -2,7 +2,6 @@
 {% import "../lib/githubLinks.html" as github -%}
 
 <article>
-  {$ github.githubLinks(doc, cliVersionInfo) $}
```
OOC, why this change?

alan-agius4 replied:
The JSON schemas no longer exist.

gkalpak replied:
We should still point people to the source imo. Where will the docs live in the future?

alan-agius4 replied:
The schemas has been replaced with a more declarative approach 

Example:
https://github.com/angular/angular-cli/blob/a4b89722633689541565be536e0b6560fb885ff2/packages/angular/cli/src/commands/config/cli.ts

https://github.com/angular/angular-cli/blob/a4b89722633689541565be536e0b6560fb885ff2/packages/angular/cli/src/commands/run/cli.ts

gkalpak replied:
So, could we fix the [realProjectRelativePath][1] to point to the new locations instead and keep the GitHub links?

[1]: https://github.com/angular/angular/blob/dc6094acdcd9bfd8409c6cb0aadd4638394f5df1/aio/tools/transforms/cli-docs-package/readers/cli-command.js#L18

alan-agius4 replied:
We could, although in most of the cases the options are a combinations of multiple files. This was also the case previously which wasn't handled as the link pointed to the build schema ex (https://github.com/angular/angular-cli/blob/master/packages/angular/cli/commands/build.json).

Example for `ng build`
- https://github.com/angular/angular-cli/blob/a4b89722633689541565be536e0b6560fb885ff2/packages/angular/cli/src/commands/build/cli.ts
-https://github.com/angular/angular-cli/blob/a4b89722633689541565be536e0b6560fb885ff2/packages/angular/cli/src/command-builder/architect-command-module.ts#L31-L44
- https://github.com/angular/angular-cli/blob/a4b89722633689541565be536e0b6560fb885ff2/packages/angular_devkit/build_angular/src/builders/browser/schema.json

In general, I don't see the edit link in the here to add much value considering the most likely the file that will be linked will be the incorrect one. As a matter of fact, I don't recall any PRs were opened from AIO.

gkalpak replied:
Fair enough. If you don't think it adds value, let's omit the links.

gkalpak reviewed (commented):
Preemptively looks reasonable :smiley:

alan-agius4 committed:
fixup! feat(docs-infra): update CLI templates to match changes with new parser and folder structure

alan-agius4 committed:
build(docs-infra): upgrade cli command docs sources to a517160a2

Updating [angular#master](https://github.com/angular/angular/tree/master) from [cli-builds#master](https://github.com/angular/cli-builds/tree/master).

alan-agius4 committed:
fixup! feat(docs-infra): update CLI templates to match changes with new parser and folder structure

mary-poppins commented:
You can preview 70c7a46 at https://pr45225-70c7a46.ngbuilds.io/.

alan-agius4 committed:
fixup! feat(docs-infra): update CLI templates to match changes with new parser and folder structure

mary-poppins commented:
You can preview 34aedc3 at https://pr45225-34aedc3.ngbuilds.io/.

alan-agius4 commented:
@gkalpak and @josephperrott kindly take a look.

gkalpak commented on aio/tools/transforms/templates/cli/cli-command.template.html:
```diff
@@ -2,15 +2,13 @@
 
   <div class="cli-body">
     {$ doc.shortDescription | marked $}
-    {$ doc.description | marked $}
```
OOC, why remove this? Won't these docs have descriptions any more?

alan-agius4 replied:
description has been replaced with shortDescription.

gkalpak replied:
🤷‍♂️

alan-agius4 replied:
😂

gkalpak commented on aio/tools/transforms/templates/cli/lib/cli.html:
```diff
@@ -71,7 +62,7 @@
       </td>
       <td><code class="no-auto-link">{$ renderOptionValues(option.type, option.enum) $}</td>
-      <td>{% if option.default !== undefined %}<code class="no-auto-link">{$ option.default $}</code>{% endif %}</td>
+      <td>{% if option.default !== undefined %}{% if option.default !== '' %}<code class="no-auto-link">{$ option.default $}</code>{% endif %}{% endif %}</td>
```
Could this be:

```suggestion
      <td>{% if option.default !== undefined and option.default !== '' %}<code class="no-auto-link">{$ option.default $}</code>{% endif %}</td>
```

alan-agius4 replied:
👍

josephperrott commented on aio/tools/transforms/cli-docs-package/processors/processCliCommands.js:
```diff
@@ -19,6 +19,20 @@ module.exports = function processCliCommands(createDocMessage) {
+
+          doc.usages = generateUsages(doc);
+
+          // Recurse if there are subcommands
```
So this only goes one level down correct, do we need to fully recurse further in?

alan-agius4 replied:
Not currently, since we only have 1 level subcommands in the Angular CLI

gkalpak reviewed (approved):
A couple of questions. Otherwise lgtm :+1:

josephperrott reviewed (approved):
LGTM

We should definitely have @gkalpak review as well though as he has more historical context and understanding.

alan-agius4 committed:
fixup! feat(docs-infra): update CLI templates to match changes with new parser and folder structure

mary-poppins commented:
You can preview 3a8da4e at https://pr45225-3a8da4e.ngbuilds.io/.

atscott commented:
This PR was merged into the repository by commit 7f043c98b7e6d277eec060361063adee5f12a7b6.
